### ISSUE TITLE
NextRequest types are wrong after nextjs v12.2.0

### ISSUE THREAD

**Issue by ThomasLarge** [nextauthjs/next-auth#5274]
### Question 💬

<img width="1629" alt="CleanShot 2022-09-01 at 15 53 26@2x" src="https://user-images.githubusercontent.com/15890942/187945312-2b8bde8b-ce71-42c3-bdad-0e3cc40e625f.png">

On the left are the new types from NextJS and on the Right are the types that `getToken` are using. When upgrading NextJS passed 12.2.0 the types become incompatible. 

<img width="1722" alt="CleanShot 2022-09-01 at 15 55 12@2x" src="https://user-images.githubusercontent.com/15890942/187945725-563f1b78-6273-45de-886e-c01cea039f4a.png">

This is the type of error when upgrading to NextJS v12.2.5

I believe the fix would be upgrading the nextJS package and making sure the types are updated. I'm more than happy to help out with this if I can be pointed in the right direction for how to get started :) 

### How to reproduce ☕️

```ts
import { getToken } from 'next-auth/jwt';
import { NextResponse, NextRequest } from 'next/server';

export const middleware = async (req: NextRequest) => {

    const token = await getToken({
      req, // Type error here
      secret: process.env.JWT_SECRET,
      secureCookie: secureCookie(),
    });

    ...

  return NextResponse.next();
};

```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by ThangHuuVu** [comment]
I believe bumping next to 12.2.5 in devDependencies should fix it. Feels a bit weird that a devDependencies causes this, so in addition, I think we need same version of next in peerDependencies too 🤔 
PR welcomes!

**Comment by balazsorban44** [comment]
Closed in #5384

**Comment by ryanelian** [comment]
Cannot use Next.js 12.3.1 with the latest Next-Auth package version anymore.

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: @accelist/nextjs-starter@1.0.0
npm ERR! Found: next@12.3.1
npm ERR! node_modules/next
npm ERR!   next@"12.3.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer next@"12.2.5" from next-auth@4.12.0
npm ERR! node_modules/next-auth
npm ERR!   next-auth@"4.12.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
```
